checkstyle-tester/launch.groovy

```diff
@@ -93,7 +93,10 @@ def generateCheckstyleReport(cliOptions) {
                     copyDir(getOsSpecificPath("$reposDir", "$repoName"), getOsSpecificPath("$srcDir", "$repoName"))
                 }
                 runMavenExecution(srcDir, excludes, checkstyleCfg, ignoreExceptions, checkstyleVersion, sevntuVersion)
-                postProcessCheckstyleReport(targetDir)
+                if (repoType != 'local') {
+                    repoUrl = getOsSpecificPath(System.properties.'user.dir', reposDir, repoName)
```

Comment:
I thought this fix was local type only. Why do we need changes for non-local?
Also why are we using the system user directory? All this is suppose to be local to the current repo.

Reply:
The fix is for local.
The original report as generated by checkstyle has paths like

`D:\Sourcecode\other\checkstyle-contribution\checkstyle-tester\src\main\java\android-launcher\src\com\android\launcher\AddAdapter.java`

The pre-PR code always replaces `checkstyle-contribution\checkstyle-tester\src\main\java` by `checkstyle-contribution\checkstyle-tester\repositories` and thus then has

`D:\Sourcecode\other\checkstyle-contribution\checkstyle-tester\repositories\android-launcher\src\com\android\launcher\AddAdapter.java`

The sense of the `local` type is to have the repo already present at any arbitrary place without copying it additionally to `repositories`, so the full path up to the repo name has to be replaced. For `local` the replacement value is the actually configured path `repoUrl`, for non-local repositories the replacement value is the respective full path inside the `repositories` directory.

Reply:
@rnveach ,

@Vampire mean that report is damaged by wrong links.
local mode does not do clone to "repositories" - https://github.com/checkstyle/contribution/commit/f7488e9d863f8abbdc11779fe7fbbd0f43c72c9c#diff-b56aba59072c339381ae8e62b6a145cdR89
so conversion of path should be done to NOT "repositories" folder, but to user defined path.
example: https://strkkk.github.io/4763/spotbugs/xref/checkstyle/contribution/checkstyle-tester/repositories/spotbugs/spotbugsTestCases/src/java/com/google/errorprone/bugpatterns/EmptyIfStatementNegativeCases.java.html#L18 (from https://strkkk.github.io/4763/)

the only problem I see is that reports that are based on "local" are not easy to share, as sources are not in "checkstyle-tester" folder. All links will be dead if report just copied to web.
We keep in repositories only required source files - https://github.com/strkkk/strkkk.github.io/tree/master/4763/spotbugs/xref/checkstyle/contribution/checkstyle-tester/repositories/spotbugs/spotbugsTestCases/src/java/com/google/errorprone/bugpatterns (not all files or original repo)
This probably should be mentioned in documentation, as attention note. So "local" mode is not designed for sharing in public web.

============

@Vampire , I am not comfortable with `System.properties.'user.dir'` can we keep code as it was before for mode `repoType != 'local` ? In this case postProcessCheckstyleReport will not do simply:
```
token: getOsSpecificPath(repoUrl),
value: getOsSpecificPath(repoDir)
```